packages/reactivity/src/computed.ts

```diff
@@ -18,24 +18,33 @@ export interface WritableComputedOptions<T> {
   set: ComputedSetter<T>
 }
 
+function isComputedGetter<T>(
+  getterOrOptions: ComputedGetter<T> | WritableComputedOptions<T>
+): getterOrOptions is ComputedGetter<T> {
+  return isFunction(getterOrOptions)
+}
+
 export function computed<T>(getter: ComputedGetter<T>): ComputedRef<T>
 export function computed<T>(
   options: WritableComputedOptions<T>
 ): WritableComputedRef<T>
 export function computed<T>(
   getterOrOptions: ComputedGetter<T> | WritableComputedOptions<T>
 ) {
-  const isReadonly = isFunction(getterOrOptions)
-  const getter = isReadonly
-    ? (getterOrOptions as ComputedGetter<T>)
-    : (getterOrOptions as WritableComputedOptions<T>).get
-  const setter = isReadonly
-    ? __DEV__
+  let getter: ComputedGetter<T>
+  let setter: ComputedSetter<T>
+
+  if (isComputedGetter(getterOrOptions)) {
```

Comment:
`isComputedGetter` is unnecessary, just `isFunction` is enough for TS to narrow the type.